### PR TITLE
Expect AssignmentExpression after 'of' in a for-of loop and inside a computed name

### LINKED ISSUE #147: for-of loop incorrectly recognizes Expression grammar for iterable expression; should be AssignmentExpression
See tc39/test262#473

For example this should be a `SyntaxError`:

```javascript
for (let x of 0, []) { }
```

### LINKED ISSUE #472: Expression shouldn't be allowed as computed property name
Chakra doesn't throw a SyntaxError for this code:
```js
({ [ 1, 2 ]: 3 });
```
but the specification defines [`ComputedPropertyName`](https://tc39.github.io/ecma262/#prod-ComputedPropertyName) as `[ AssignmentExpression ]`

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Fixes #147
Fixes #472

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
This pr is ready to be reviewd

File: lib/Parser/Parse.cpp [hunk 1]
```diff
@@ -3687,7 +3687,7 @@ template<bool buildAST> void Parser::ParseComputedName(ParseNodePtr* ppnodeName,
         *ppFullNameHint = FormatPropertyString(*ppNameHint, pnodeNameExpr, pNameLength, pShortNameOffset);
     }
 
-    ChkCurTokNoScan(tkRBrack, ERRsyntax);
+    ChkCurTokNoScan(tkRBrack, ERRnoRbrack);
```
Comment (ianwjhalliday) [inline lib/Parser/Parse.cpp]: Nice

File: test/es6/objlit.js [hunk 2]
```diff
@@ -123,7 +115,9 @@ var tests = [
             };
             assert.areEqual(protoObj.abc, 123, "__proto__ get assigned when used as a normal production");
 
-            assert.throws(function () { eval("var b = { ['str'] }"); },            SyntaxError, "Invalid computed identifier shorthand");
+            assert.throws(function () { eval("var b = { ['str'] }"); }, SyntaxError, "Invalid computed identifier shorthand");
```
Comment (ianwjhalliday) [inline test/es6/objlit.js]: Would you mind adding the error message argument to this `assert.throws`?

File: test/es6/objlit.js [hunk 3]
```diff
@@ -75,36 +75,28 @@ var tests = [
             var obj = {
                 ["foo" + "bar"]         : 1,
                 [1 * 10 * 10]           : 2,
-                ["foobar", "notfoobar"] : 3, // Should evaluate to notfoobar
```
Comment (ianwjhalliday) [inline test/es6/objlit.js]: To ensure we're parsing the right starting grammar production we should have a test that uses one of the production rules that are unique to that production.  Simplest would be a simple assignment, so something like
`[a = 'baz']: 3,`.

Here and the other cases below.

**Comment by ianwjhalliday** [comment]
Just a couple minor comments on the tests.  Looks great, nice a simple.  Thanks for taking the time to help us knock this issue off the list.

**Comment by nicolo-ribaudo** [comment]
@ianwjhalliday I updated the tests as you asked

**Comment by ianwjhalliday** [comment]
Sorry for the delay.  Looks great!  I'll get this merged in later today if I can.

**Comment by ianwjhalliday** [comment]
@nicolo-ribaudo thanks again!

**Pull request merged**